immunopepper/filter.py

```diff
@@ -0,0 +1,202 @@
+"""Contains the functionality used for filtering"""
+import sys
+import re
+
+import numpy as np
+from functools import reduce
+
+from .constant import NOT_EXIST
+from .namedtuples import OutputBackground
+
+def _collect_remove_ids(exon_junction_dict):
+    """ Get an auxiliary dictionary used for filtering. Assigns to each junction all overlapping exon pairs.
+
+    Parameters
+    ----------
+    exon_junction_dict: dict. (read_frame, pos_mid_1, pos_mid_2, variant_combination)
+                               -> List[Tuple(output_idx, pos_start, pos_end,)]
+
+    Returns
+    -------
+    remove_id_list: List[junction ids to remove]
+    """
+
+    # TODO: [Warning!] if two output lines have identical
+    # coordinates and readframe both of them will be removed
+    remove_id_list = []
+    removed = set()
+    for exon_pair_list in exon_junction_dict.values():
+        L = len(exon_pair_list)
+        if L < 2:
+            continue
+        for i, exon_pair in enumerate(exon_pair_list):
+            if exon_pair[0] in removed:
+                continue
+            for j in range(i + 1, L):
+                if exon_pair_list[j][0] in removed:
+                    continue
+                i_pos1 = exon_pair[1]
+                i_pos2 = exon_pair[2]
+                j_pos1 = exon_pair_list[j][1]
+                j_pos2 = exon_pair_list[j][2]
+                if i_pos1 >= j_pos1 and i_pos2 <= j_pos2:
+                    remove_id_list.append(exon_pair[0])
+                    removed.add(exon_pair[0])
+                    break
+                if j_pos1 >= i_pos1 and j_pos2 <= i_pos2:
+                    remove_id_list.append(exon_pairs[j][0])
```

Comment:
exon_pairs is unresolved, pycharm tells me

Reply:
good catch. fixed.